### PR TITLE
[Luxtronik] Fix possible out ouf bound error for older heatpumps

### PR DESCRIPTION
**Pull request by sgiehl**
As reported on the forum some older heatpumps seem to serve less than 154 socket channels, causing a out of bounds error.

See https://community.openhab.org/t/migration-of-novelan-luxtronic-binding-to-openhab2/70743/132?u=sgiehl

### CODE REVIEW

File: bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/HeatpumpConnector.java [hunk 1]
```diff
@@ -71,7 +71,7 @@ public void read() throws UnknownHostException, IOException {
             // the thermal energies can be unreasonably high in some cases, probably due to a sign bug in the firmware
             // trying to correct this issue here
             for (int i = 151; i <= 154; i++) {
-                if (heatpumpValues[i] >= 214748364) {
+                if (heatpumpValues.length >= i && heatpumpValues[i] >= 214748364) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/HeatpumpConnector.java:74]: I guess it should rather be " > i" ?

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/HeatpumpConnector.java (hunk 1 above)
Comment (sgiehl) [reply bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/HeatpumpConnector.java:74]: Sure.

**Comment by lolodomo** [comment]
Please sign-off your PR.

**Comment by sgiehl** [comment]
@lolodomo sorry for that. Had been trying to fix that on another machine, so the signoff didn't match :see_no_evil:  Should be fixed now.

**Review by lolodomo — APPROVED** [review]

**Pull request merged**